src/test/java/uk/gov/hmcts/reform/pbis/EmailServiceTest.java

```diff
@@ -0,0 +1,112 @@
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.pbis.model.EmailToSend;
+import uk.gov.hmcts.reform.pbis.model.PrivateBetaRegistration;
+import uk.gov.hmcts.reform.pbis.notify.NotificationClientProvider;
+import uk.gov.service.notify.NotificationClientApi;
+import uk.gov.service.notify.NotificationClientException;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class EmailServiceTest {
+
+    private EmailService emailService;
+
+    @Mock
+    private NotificationClientApi notificationClient;
+
+    @Mock
+    private NotificationClientProvider notificationClientProvider;
+
+    @Mock
+    private EmailCreator emailCreator;
+
+    private final PrivateBetaRegistration privateBetaRegistration = new PrivateBetaRegistration(
+        "reference id 123",
+        "service 123",
+        "email@example.com",
+        "first name",
+        "last name"
+    );
+
+    private final EmailToSend emailToSend = new EmailToSend(
+        "email@example.com",
+        "template id 123",
+        ImmutableMap.of("name", "value"),
+        "reference id 123"
+    );
+
+
+    @Before
+    public void setup() {
+        when(notificationClientProvider.getClient(anyString())).thenReturn(notificationClient);
+        emailService = new EmailService(notificationClientProvider, emailCreator);
+    }
+
+    @Test
+    public void sendWelcomeEmail_should_call_email_client_with_email_data()
+        throws NotificationClientException {
+
+        when(emailCreator.createEmailToSend(privateBetaRegistration)).thenReturn(emailToSend);
+
+        emailService.sendWelcomeEmail(privateBetaRegistration);
+
+        verify(emailCreator).createEmailToSend(privateBetaRegistration);
+        verify(notificationClientProvider).getClient(privateBetaRegistration.service);
+
+        verify(notificationClient).sendEmail(
+            emailToSend.templateId,
+            emailToSend.emailAddress,
+            emailToSend.templateFields,
+            emailToSend.referenceId
+        );
+
+        verifyNoMoreInteractions(emailCreator, notificationClient, notificationClientProvider);
+    }
+
+    @Test(expected = EmailSendingException.class)
+    public void sendWelcomeEmail_should_throw_exception_when_email_creator_fails() {
+        doThrow(new EmailSendingException("test exception", null))
+            .when(emailCreator)
+            .createEmailToSend(any());
+
+        emailService.sendWelcomeEmail(privateBetaRegistration);
+    }
+
+    @Test(expected = EmailSendingException.class)
+    public void sendWelcomeEmail_should_throw_exception_when_notification_client_fails()
+        throws NotificationClientException {
+
+        doThrow(mock(NotificationClientException.class))
+            .when(notificationClient)
+            .sendEmail(anyString(), anyString(), anyMap(), anyString());
+
+        emailService.sendWelcomeEmail(privateBetaRegistration);
+    }
+
+    @Test(expected = EmailSendingException.class)
```

Comment:
Did you consider assertJ way of testing exceptions?

Not insisting on it, but check it out if you haven't.

Reply:
I didn't know about `catchThrowable` when writing this. Will change.